metashare/api/views.py

```diff
@@ -209,6 +217,15 @@ def perform_create(self, *args, **kwargs):
     def perform_destroy(self, instance):
         instance.queue_delete(originating_user_id=str(self.request.user.id))
 
+    def destroy(self, request, *args, **kwargs):
+        scratch_org = self.get_object()
+        if not request.user == scratch_org.owner:
+            return Response(
+                {"error": _("Requesting user did not create scratch org.")},
+                status=status.HTTP_403_FORBIDDEN,
+            )
+        return super().destroy(request, *args, **kwargs)
```

Comment:
Honestly, I like option 2 more, as I think it'll clear the way for option 1, and smooth the API surface.

Reply:
So, the current logic changes the `ScratchOrg.owner`, right? I think I need to walk through with you what it's currently doing, because looking at the backend, I'm confused about how it's even working now, given my assumptions—so I doubt my assumptions.

Reply:
No, the current logic first deletes the org, then does the task reassignment -- all controlled from the front-end. We're never changing the org.owner -- we discussed it but decided it had other issues. (Ping me if you'd like to talk about this more!)

Reply:
OK, so we should, on the backend:
 - When there's a PUT to a Task,
 - If it changes the `assigned_*` attribute,
 - Delete the attached org
 - And create a new one,
 - Returning a 200, but later sending a websocket event about the new org.

Does that sound right?

Reply:
Nope. :) We're not talking about creating a new org at all here -- only deleting the existing one when the role is re-assigned.

Reply:
Done, as per that!